### ISSUE TITLE
Issues with printing languages in unicode.

### ISSUE THREAD

**Issue by jsphonorio** [kivy/kivy#2669]
I am trying to print in Tamil and running into an issue where Kivy misprints the unicode language. Here is a short script and an Image showing the issue. To show comparison, here is same text print in Pyside and how it looks correctly.

```
  #!/usr/bin/python
  # -*- coding: utf-8 -*-

  from kivy.app import App
  from kivy.uix.button import Button

  class TestApp(App):
      def build(self):
          return Button(text='விகடகவி'.decode('utf-8'),font_name='Uni Ila.Sundaram-01.ttf',font_size=50)

  TestApp().run()
```

![kivy-example](https://cloud.githubusercontent.com/assets/9734945/5040671/a741f07a-6bf5-11e4-9a2b-2aca504b953e.png)
![pyside-example](https://cloud.githubusercontent.com/assets/9734945/5040672/abc1b8ce-6bf5-11e4-8f29-ffc73a3ab1a8.png)

The language combines letters together and sometimes rearranges consonants around vowels. Kivy has trouble parsing this.

These last two pictures just provide another example of a different phrase on a different computer. Same issue.

![kivy-example2](https://cloud.githubusercontent.com/assets/9734945/5040712/a4aac584-6bf6-11e4-9dc3-a4647d6809ee.png)

![pyside-example2](https://cloud.githubusercontent.com/assets/9734945/5040713/aa0fa27e-6bf6-11e4-81de-122b0c016aca.png)
##

**Comment by pyprof** [comment]
I am facing same problem. But to write Bengali. Please be refereed to the screenshot to see how Kivy is breaking combined letters.

Here is the code.... https://bpaste.net/show/92402933377e
Here are the screenshots... 
- I am expecting something like this....
![a](https://cloud.githubusercontent.com/assets/12394079/7915229/0a6ac65a-089e-11e5-84f6-51d77de8c8d5.png)

- Getting something like this...
![screenshot from 2015-06-01 20 30 57](https://cloud.githubusercontent.com/assets/12394079/7915274/478eded6-089e-11e5-9b99-34e94fae05fa.png)

**Comment by chintal** [comment]
As an additional data point, if it helps, vanilla pillow (PIL) Image, ImageDraw, and ImageFont have the same rendering problems as kivy. However, with libraqm also installed, pillow's renders are correct. Kivy's renders are not changed when libraqm is installed.

**Comment by chintal** [comment]
Kivy with the pango text backend works fine. In the languages I can read, there aren't any obvious errors. Some mention of the pango backend would be nice to have with reference to the text core, specifically with some mention of the downsides or risks of the experimental backend, if there are any, and that the pango backend will likely produce better results with unicode fonts for complex scripts.

I have tried the following text render routes :
 - kivy, default : output is obviously incorrect
 - pillow, default : output is obviously incorrect
 - pillow, libraqm : output is sort of correct, but there are some issues, for ex. with kannada
 - pyvips : output is actually probably the most accurate with this one, and almost efortlessly. It seems to be falling back to system fonts or so by itself when necessary, unlike all the others. The code itself is pretty strange though, and IMO a little too complicated to use on a regular basis. 
 - kivy, pango : output looks fine. It probably has the best results across a series of languages, including pyvips if you factor in the extra headache of dealing with pyvips code.

Here's a screenshot of the test code. The kivy column is rendered normally by the application in Label widgets, all the others are images generated by those libraries and displayed by kivy Image widgets.

The languages marked with a * are ones where I'm not certain the source string is 'correct'. In practice, punjabi and marathi are probably correct as well, and I have no idea about the ones after that.
 
![Screenshot_20190901_140317](https://user-images.githubusercontent.com/226507/64073863-75494000-ccc1-11e9-8d2e-bcf1ca9d61f0.png)

The code that was used to generate the image is [here](https://github.com/chintal/indic-fonts-test).

**Comment by tripleee** [comment]
What's the status of this now? Seems to be pending on #4902 which however also seems stale. Tangentially, another data point at https://stackoverflow.com/q/63047020/874188

**Comment by MS-Jahan** [comment]
@pyprof  I also need a simple solution for Bangla.

**Comment by MS-Jahan** [comment]
@pyprof Maybe we can use Bangla text after converting to ANSI and usng [font] tag.

https://github.com/MS-Jahan/unicode2bijoy

**Comment by Sahil-pixel** [comment]
is there Any solutions ? I am facing same problem with Bengali language .

**Comment by MS-Jahan** [comment]
> is there Any solutions ? I am facing same problem with Bengali language .

You can convert unicode Bangla text to ANSI and show them in Kivy. See here https://github.com/MS-Jahan/unicode2bijoy

**Comment by Sahil-pixel** [comment]
@MS-Jahan 
I want other languages also....not only bengali .

**Comment by MS-Jahan** [comment]
> @MS-Jahan
> I want other languages also....not only bengali .

Well, that wasn't a perfect solution. You can also try searching for Unicode to ANSI converter for other languages or create your own. That'll be a temporary solution before we can get any solution from the Kivy Team, I guess.

**Comment by Sahil-pixel** [comment]
@MS-Jahan  This issue is open from 2014 .I don't know why kivy team not solving this problem.

**Comment by matham** [comment]
As mentioned [here](https://github.com/kivy/kivy/issues/7149), this should be fixed once sdl_ttf is released (or someone undertakes to build kivy using the unreleased version of sdl_ttf) because it now can use harfbuzz.

**Comment by Julian-O** [comment]
These issues sound very similar to #7227.

It was solved with a new feature added in Kivy 2.2.0 for users of TTF fonts and SDL to display it: [Label's font_script_name](https://kivy.org/doc/stable/api-kivy.uix.label.html#kivy.uix.label.Label.font_script_name)

I suspect it will solve these problems too - at least in applications where you know that the language will be.

Could any of the people who complained about this problem (@jsphonorio @pyprof @MS-Jahan @Sahil-pixel ) please try setting that value, and telling us if the problem is gone?
